aclmapping/bank/mappings.go

```diff
@@ -0,0 +1,81 @@
+package aclbankmapping
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkacltypes "github.com/cosmos/cosmos-sdk/types/accesscontrol"
+	aclkeeper "github.com/cosmos/cosmos-sdk/x/accesscontrol/keeper"
+	acltypes "github.com/cosmos/cosmos-sdk/x/accesscontrol/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	utils "github.com/sei-protocol/sei-chain/aclmapping/utils"
+)
+
+var ErrorInvalidMsgType = fmt.Errorf("invalid message received for bank module")
+
+func GetBankDepedencyGenerator() aclkeeper.DependencyGeneratorMap {
+	dependencyGeneratorMap := make(aclkeeper.DependencyGeneratorMap)
+
+	// dex place orders
+	placeOrdersKey := acltypes.GenerateMessageKey(&banktypes.MsgSend{})
+	dependencyGeneratorMap[placeOrdersKey] = MsgSendDependencyGenerator
+
+	return dependencyGeneratorMap
+}
+
+func MsgSendDependencyGenerator(keeper aclkeeper.Keeper, ctx sdk.Context, msg sdk.Msg) ([]sdkacltypes.AccessOperation, error) {
+	msgSend, ok := msg.(*banktypes.MsgSend)
+	if !ok {
+		return []sdkacltypes.AccessOperation{}, ErrorInvalidMsgType
+	}
+
+	accessOperations := []sdkacltypes.AccessOperation{
+		// MsgSend also checks if the coin denom is enabled, but the information is from the params.
+		// Changing the param would require a gov proposal, which is synchrounos by default
+
+		// Checks balance of sender
+		{
+			AccessType:         sdkacltypes.AccessType_READ,
+			ResourceType:       sdkacltypes.ResourceType_KV,
+			IdentifierTemplate: utils.GetIdentifierTemplatePerModule(utils.BANK, msgSend.FromAddress),
```

Comment:
instead of having the identifier template be per module, we should create more resource types to further segment the population. For example, under KV, we can have resource type `KV_BANK`, and then `KV_BANK_BALANCE` and `KV_BANK_PARAMS`, and then for the specific account we could have another resource type called `KV_BANK_BALANCE_ACCOUNT`. This way its much easier to have static dependencies that are granular enough to still allow for parallelization, and then the static read for this would be resource `KV_BANK_BALANCE` whereas the dynamic resource would be `KV_BANK_BALANCE_ACCOUNT` with the account identifier. We can represent this hierarchy in the resource hierarchy tree in resource.go in `sei-cosmos`.

Reply:
@BrandonWeng lets leave a TODO at the top of the generator function to make the resources more granular down the road

Reply:
Yessir, thanks! As discussed offline - we want to unblock the work first and this can be something we address later once we get the concurrency working